gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/TaskService.java

```diff
@@ -148,40 +150,71 @@ void addProcessEvent(final Task task,
-                .limit(ChronoUnit.HOURS.between(savedProcessFile.getStartingAvailabilityDate(), savedProcessFile.getEndingAvailabilityDate()))
-                .parallel()
-                .map(this::getTaskWithStatusUpdate) //by default statusUpdated false except if task is created
-                .collect(Collectors.toSet());
+    public Set<TaskWithStatusUpdate> addProcessFileToTasks(final ProcessFile savedProcessFile,
+                                                           final FileEventType fileEventType,
+                                                           final boolean isInput,
+                                                           final boolean withStatusUpdate) {
+        final OffsetDateTime start = savedProcessFile.getStartingAvailabilityDate();
+        final OffsetDateTime end = savedProcessFile.getEndingAvailabilityDate();
+        final Set<TaskWithStatusUpdate> allTasks = new HashSet<>();
 
-        for (TaskWithStatusUpdate taskWithStatusUpdate : taskWithStatusUpdateSet) {
-            Task task = taskWithStatusUpdate.getTask();
-            addFileEventToTask(task, fileEventType, savedProcessFile);
-            if (isInput) {
-                removeUnavailableProcessFileFromTaskRunHistory(savedProcessFile, task, fileEventType);
-            }
-            task.addProcessFile(savedProcessFile);
-            if (withStatusUpdate && !taskWithStatusUpdate.isStatusUpdated() && isInput) {
-                //if taskWithStatusUpdate is already false and the process file of type input, we need to check if the status should be updated
-                boolean statusUpdateDueToFileArrival = checkAndUpdateTaskStatus(task, true);
+        //Deal with existing tasks in database
+        final Map<OffsetDateTime, TaskWithStatusUpdate> existingTasksInDatabase = taskRepository.findAllByTimestampBetween(start, end)
+                .stream()
+                .collect(Collectors.toMap(Task::getTimestamp, task -> new TaskWithStatusUpdate(task, false)));
+
+        existingTasksInDatabase.values().forEach(taskWithStatusUpdate -> {
+            applyProcessFileToTask(savedProcessFile, fileEventType, isInput, taskWithStatusUpdate);
+            if (withStatusUpdate && isInput) {
+                final Task task = taskWithStatusUpdate.getTask();
+                final boolean statusUpdateDueToFileArrival = checkAndUpdateTaskStatus(task, true);
                 taskWithStatusUpdate.setStatusUpdated(statusUpdateDueToFileArrival);
+
                 if (statusUpdateDueToFileArrival) {
-                    LOGGER.info("Update task status of task with timestamp {} when processFile {} arrived to status {}", task.getTimestamp(), savedProcessFile.getFilename(), task.getStatus());
+                    LOGGER.info("Update task status of task with timestamp {} when processFile {} arrived to status {}",
```

Comment:
Didn't notice it before, but it sounds curious to say "Update **task** status of **task** with ...". Removing the first "task" looks better: "Update status of task with ...".